### ISSUE TITLE
tests fail with "unknown option command"

### ISSUE THREAD

**Issue by peff** [jonas/tig#912]
Running `make test` I get failures for almost every test with output like this:

```
--- a/expected/noprefix-conflict.stderr
+++ b/noprefix-conflict.stderr
@@ -0,0 +1,22 @@
+tig warning: <built-in>:231: Unknown option command: branch.current=main-head
+tig warning: <built-in>:232: Unknown option command: branch.local=main-ref
+tig warning: <built-in>:233: Unknown option command: branch.plain=main-ref
+tig warning: <built-in>:234: Unknown option command: branch.remote=main-remote
+tig warning: <built-in>:236: Unknown option command: diff.meta=diff-header
+tig warning: <built-in>:237: Unknown option command: diff.meta=diff-index
+tig warning: <built-in>:238: Unknown option command: diff.meta=diff-oldmode
+tig warning: <built-in>:239: Unknown option command: diff.meta=diff-newmode
+tig warning: <built-in>:240: Unknown option command: diff.frag=diff-chunk
+tig warning: <built-in>:241: Unknown option command: diff.old=diff-del
+tig warning: <built-in>:242: Unknown option command: diff.new=diff-add
+tig warning: <built-in>:244: Unknown option command: diff-highlight.oldHighlight=diff-del-highlight
+tig warning: <built-in>:245: Unknown option command: diff-highlight.newHighlight=diff-add-highlight
+tig warning: <built-in>:247: Unknown option command: grep.filename=grep.file
+tig warning: <built-in>:248: Unknown option command: grep.linenumber=grep.line-number
+tig warning: <built-in>:249: Unknown option command: grep.separator=grep.delimiter
+tig warning: <built-in>:251: Unknown option command: status.branch=status.header
+tig warning: <built-in>:252: Unknown option command: status.added=stat-staged
+tig warning: <built-in>:253: Unknown option command: status.updated=stat-staged
+tig warning: <built-in>:254: Unknown option command: status.changed=stat-unstaged
+tig warning: <built-in>:255: Unknown option command: status.untracked=stat-untracked
+tig: Failed to load user config.: Error in built-in config
```

which makes sense. Those come from builtin-config.c, which is generated from tigrc. The latter has a bunch of backslash-escaped lines:

```
set git-colors = \
  branch.current=main-head \
[etc...]
```

but the backslashes are eaten. This makes sense because we pass the data through `read` in https://github.com/jonas/tig/blob/e8d5546451eb4d2937ac940d6818708225bebd36/tools/make-builtin-config.sh#L33 when we probably ought to be using `read -r` to preserve backslash sequences.

But I'm not sure why it works in Travis, or why just running tig doesn't complain in the same way. Maybe because those use an installed tigrc, and the builtin one has just always been broken since we added `git-colors` in 6c58ecc3a29e7dd0387ca3dc241d7c43d5cfaf3a? That seems like a long time for nobody to have noticed.

I'm happy to prepare a patch to `make-builtin-config.sh`, but I have a feeling I'm missing something. Has anybody seen this before?

**Comment by peff** [comment]
> we probably ought to be using read -r to preserve backslash sequences

Hrm, but that's not quite the whole story. We do add an extra layer of backslash escaping (on top of what is needed for the C escaping) in https://github.com/jonas/tig/blob/e8d5546451eb4d2937ac940d6818708225bebd36/tools/make-builtin-config.sh#L24. But for some reason that doesn't work for me; maybe a different version of sed or something? Or maybe my shell `read` (which is just dash from Debian) handles things differently?

If I halve the number of backslashes in the `sed` invocation and then use `read -r`, it does make the problem go away. Puzzling...

**Comment by peff** [comment]
Should be fixed by #914
